tree/dataframe/inc/ROOT/RDF/RDefaultFor.hxx

```diff
@@ -0,0 +1,159 @@
+        fLastResults(lm.GetNSlots() * RDFInternal::CacheLineStep<T>()),
+        fValues(lm.GetNSlots())
+   {
+      fLoopManager->Register(this);
+      // We suppress errors that TTreeReader prints regarding the missing branch
+      fLoopManager->GetSuppressErrorsForMissingBranches().push_back(fColumnNames[0]);
+   }
+
+   RDefaultFor(const RDefaultFor &) = delete;
+   RDefaultFor &operator=(const RDefaultFor &) = delete;
+   RDefaultFor(RDefaultFor &&) = delete;
+   RDefaultFor &operator=(RDefaultFor &&) = delete;
+   ~RDefaultFor()
+   {
+      fLoopManager->Deregister(this);
+      ROOT::Internal::RDF::Erase(fColumnNames[0], fLoopManager->GetSuppressErrorsForMissingBranches());
+   }
+
+   void InitSlot(TTreeReader *r, unsigned int slot) final
+   {
+      fValues[slot] = RDFInternal::GetColumnReader<T>(
+         slot, fColRegister.GetReader(slot, fColumnNames[0], fVariation, typeid(T)), *fLoopManager, r, fColumnNames[0]);
+      fLastCheckedEntry[slot * RDFInternal::CacheLineStep<Long64_t>()] = -1;
+   }
+
+   /// Return the (type-erased) address of the Define'd value for the given processing slot.
+   void *GetValuePtr(unsigned int slot) final
+   {
+      return static_cast<void *>(&fLastResults[slot * RDFInternal::CacheLineStep<T>()]);
+   }
+
+   /// Update the value at the address returned by GetValuePtr with the content corresponding to the given entry
+   void Update(unsigned int slot, Long64_t entry) final
+   {
+      if (entry != fLastCheckedEntry[slot * RDFInternal::CacheLineStep<Long64_t>()]) {
+         // evaluate this define expression, cache the result
+         fLastResults[slot * RDFInternal::CacheLineStep<T>()] = GetOrDefault(slot, entry);
+         (void)entry; // avoid unused parameter warning (gcc 12.1)
+         fLastCheckedEntry[slot * RDFInternal::CacheLineStep<Long64_t>()] = entry;
+      }
+   }
+
+   void Update(unsigned int /*slot*/, const ROOT::RDF::RSampleInfo & /*id*/) final {}
+
+   const std::type_info &GetTypeId() const final { return typeid(T); }
+
+   /// Clean-up operations to be performed at the end of a task.
+   void FinalizeSlot(unsigned int slot) final
+   {
+      fValues[slot] = nullptr;
+
+      for (auto &e : fVariedDefines)
+         e.second->FinalizeSlot(slot);
+   }
+
+   /// Create clones of this Define that work with values in varied "universes".
+   void MakeVariations(const std::vector<std::string> &variations) final
+   {
+      for (const auto &variation : variations) {
+         if (std::find(fVariationDeps.begin(), fVariationDeps.end(), variation) == fVariationDeps.end()) {
+            // this Defined quantity does not depend on this variation, so no need to create a varied RDefine
+            continue;
+         }
+         if (fVariedDefines.find(variation) != fVariedDefines.end())
+            continue; // we already have this variation stored
+
+         // the varied defines get a copy of the callable object.
+         // TODO document this
```

Comment:
Actually I don't even think this comment holds here because this wording is a leftover copy-paste from the implementation of RDefine: https://github.com/root-project/root/blob/dda3ac9a474f438e7099492d37d435abaa23c606/tree/dataframe/inc/ROOT/RDF/RDefine.hxx#L159

In the case of RDefaultFor, there is no callable to copy, hence I will just remove this comment

Reply:
Although I agree that we should not forget about the systematic variations use case, I will write some unit testing more specific to that.

Reply:
In the last commit I added a first unit test with a simple variation on a branch with collections (std::vector<int>)